product_tax_multicompany_default/models/product_template.py

```diff
@@ -0,0 +1,57 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Carlos Dauden - Tecnativa <carlos.dauden@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from openerp import api, models
+
+
+class ProductTemplate(models.Model):
+    _inherit = 'product.template'
+
+    def taxes_by_company(self, field, company_id, match_tax_ids=None):
+        taxes_ids = []
+        if match_tax_ids is None:
+            taxes_ids = self.env['ir.values'].get_default(
+                'product.template', field, company_id=company_id)
+        # If None: return default taxes if []: return empty list
+        if not match_tax_ids:
+            return isinstance(taxes_ids, list) and taxes_ids or []
+        AccountTax = self.env['account.tax']
+        for tax in AccountTax.browse(match_tax_ids):
+            taxes_ids.extend(AccountTax.search([
+                ('name', '=', tax.name),
+                ('company_id', '=', company_id)
+            ]).ids)
+        return taxes_ids
+
+    @api.multi
+    def set_multicompany_taxes(self):
+        self.ensure_one()
+        tax_ids = self.taxes_id.ids
+        supplier_tax_ids = self.supplier_taxes_id.ids
+        company_id = self.env.user.company_id.id
+
+        self = self.sudo()
+        default_taxes = self.taxes_by_company('taxes_id', company_id)
```

Comment:
If these are a list of IDs, call it `default_tax_ids`, or better `default_customer_tax_ids`